dask/dataframe/groupby.py

```diff
@@ -520,6 +553,7 @@ def _is_grouped_by_sliced_column(self, df, index):
         """
         Return whether index is a Series sliced from df
         """
+        # TODO: handle list of series
```

Comment:
While trying to figure out this todo. If noticed that with the step of index normalization quite some code could be simplified. I adapted the creation of `_meta` and `_meta_nonmepty` accordingly. I also added tests for the content of the `_meta` objects.